src/main/java/mate/academy/dao/UserDao.java

```diff
@@ -0,0 +1,11 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.User;
+
+public interface UserDao {
+    User create(User user) throws DataProcessingException;
```

Comment:
I think it's needless 
```suggestion
    User create(User user);
```